parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetInputSplit.java

```diff
@@ -222,6 +227,14 @@ public long[] getRowGroupOffsets() {
     return rowGroupOffsets;
   }
 
+  public ParquetMetadata getFooter() {
+    return footer;
+  }
+
+  public void setFooter(ParquetMetadata footer) {
```

Comment:
The `ParquetInputSplit` is marked as deprecated. And the recommended usage is `FileSplit`.  How does Spark set the `footer` after the `ParquetInputSplit` removed?

Reply:
Now parquet will convert the input split to `ParquetInputSplit` and build the reader with it. I think if  `ParquetInputSplit` was removed from ParquetFileReader class, spark need a shim class to work with different parquet version. 

That will be a big change.

Reply:
I think this PR can be a good reason to push the spark community to migrate. Or we can fix this in only spark 4.x?

Reply:
Already filed a WIP ticket https://github.com/apache/spark/pull/44853 for spark 4 and will discuss about this change in spark side in that PR after this PR is merged.

Reply:
IIUC, other comments have suggested that we should not work on a deprecated interface. Therefore I don't expect this PR will be merged as is. It would be good to figure out the final solution on the spark side before any action here.